File: frameowkrs/java-selenium-rest-assured/pages/CheckoutPage.java
package pages;

import java.util.regex.Matcher;
import java.util.regex.Pattern;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;

public final class CheckoutPage extends BasePage {
    public CheckoutPage(WebDriver driver) { super(driver); }

    public int placeOrder() {
        visible("checkout-use-fake-name-email").click();
        visible("checkout-use-fake-address").click();
        visible("checkout-use-testing-card").click();
        visible("checkout-submit").click();
        wait.until(ExpectedConditions.urlMatches("/orders/\\d+$"));
        Matcher match = Pattern.compile("/orders/(\\d+)$").matcher(driver.getCurrentUrl());
        if (!match.find()) throw new AssertionError("Order ID was not present in the order detail URL");
        return Integer.parseInt(match.group(1));
    }
}

File: frameworks/typescript-cypress/pages/checkout_page.ts
import { ACTIONS, step } from '../helpers/steps';

export class CheckoutPage {
  fillTestingDetails(): void {
    step(ACTIONS.checkoutFill, () => {
      cy.get('[data-testid="checkout-use-fake-name-email"]').check();
      cy.get('[data-testid="checkout-use-fake-address"]').check();
      cy.get('[data-testid="checkout-use-testing-card"]').check();
    });
  }

  submitOrder(): Cypress.Chainable<number> {
    return step(ACTIONS.checkoutSubmit, () => {
      cy.get('[data-testid="checkout-submit"]').click();
      return cy
        .url()
        .should('match', /\/orders\/\d+$/)
        .then((url) => {
          const orderId = Number(url.match(/\/orders\/(\d+)$/)?.[1]);
          if (!orderId) throw new Error('Order ID was not present in the order detail URL');
          return orderId;
        });
    });
  }

  placeOrder(): Cypress.Chainable<number> {
    this.fillTestingDetails();
    return this.submitOrder();
  }
}
